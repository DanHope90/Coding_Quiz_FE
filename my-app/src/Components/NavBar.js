import React from "react";
import { Link } from "react-router-dom";
import "../Styles/NavBar.css";
import { FaHome, FaSave, FaRocket } from "react-icons/fa";
import { BiTestTube } from "react-icons/bi";
import questionmark from "../Images/question-mark.jpg";

function NavBar() {
  return (
    <div className="navbar">
      <ul className="navbar-links">
        <li className="q-mark-pic">
          <img src={questionmark} alt="question mark" />
        </li>
        <li className="navbar-links-item">
          <Link className="navbar-links-link" to="/">
            Home
            {' '}
            <FaHome />
          </Link>
        </li>
        <li className="navbar-links-item">
          <Link className="navbar-links-link" to="/take-a-quiz">Take A Quiz <FaRocket /></Link>
        </li>
        <li className="navbar-links-item">
          <Link className="navbar-links-link" to="/saved-quizzes">Saved Quizzes <FaSave /></Link>
        </li>
        <li className="navbar-links-item">
          <Link className="navbar-links-link" to="/test-quiz">Test Quiz on JavaScript <BiTestTube /></Link>
        </li>
      </ul>
    </div>
  );
}

export default NavBar;
