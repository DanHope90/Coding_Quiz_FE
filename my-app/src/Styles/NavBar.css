.navbar {
    background-color: rgb(186, 185, 185);
    border-bottom: 2px solid black;
}

.navbar-links {
    margin: 0;
    padding: 0;
    display: flex;
    align-items: center;
}

.navbar-links-item {
    list-style-type: none;
    display: block;
}

.navbar-links-link {
    color: black;
    text-decoration: none;
    font-weight: bold;
    margin: 0;
    padding: 5px;
    display: inline-block;
}

.navbar-links-link:active{
    color:white;
}

.q-mark-pic {
    display: flex;
    justify-content: flex-end;
    align-items: flex-end;
    font-size: 70px;
}

img {
    width: inherit;
    height: inherit;
}

@media only screen and (max-width: 600px) {
    .navbar{
        height: 50px;
        font-size: 10px;
        border-bottom: none;
    }
    .q-mark-pic{
        /* On a smaller device, the question mark icon does not render
        Necessary to save space on mobile phones */
        font-size: 0px;
    }
}