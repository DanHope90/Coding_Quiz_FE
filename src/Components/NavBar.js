import React from "react";
import { Link } from "react-router-dom";
import "../Styles/NavBar.css";
import { FaHome, FaSave, FaRocket } from "react-icons/fa";
import { BiTestTube } from "react-icons/bi";
import { BsQuestionSquareFill } from "react-icons/bs";
import { MdLeaderboard } from "react-icons/md";

function NavBar() {
  return (
    <div className="navbar">
      <ul className="navbar-links">
        <li className="q-mark-pic">
          <BsQuestionSquareFill />
        </li>
        <li className="navbar-links-item">
          <Link className="navbar-links-link" to="/">Home <FaHome /></Link>
        </li>
        <li className="navbar-links-item">
          <Link className="navbar-links-link" to="/take-a-quiz">Take Quiz <FaRocket /></Link>
        </li>
        <li className="navbar-links-item">
          <Link className="navbar-links-link" to="/saved-quizzes">Saved Quizzes <FaSave /></Link>
        </li>
        <li className="navbar-links-item">
          <Link className="navbar-links-link" to="/random-quiz">Random Quiz <BiTestTube /></Link>
        </li>
        <li className="navbar-links-item">
          <Link className="navbar-links-link" to="/leaderboard">Leaderboard <MdLeaderboard /></Link>
        </li>
      </ul>
    </div>
  );
}

export default NavBar;
