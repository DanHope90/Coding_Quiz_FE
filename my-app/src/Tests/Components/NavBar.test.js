import NavBar from "../../Components/NavBar";
import React from "react";
import renderer from "react-test-renderer";
import {BrowserRouter} from "react-router-dom";

test("renders the NavBar correctly", () => {
    const testRenderer = renderer.create(<BrowserRouter><NavBar  /></BrowserRouter>);
    expect(testRenderer).toMatchSnapshot();
});
