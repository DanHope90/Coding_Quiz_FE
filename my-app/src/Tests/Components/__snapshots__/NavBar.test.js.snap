// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the NavBar correctly 1`] = `
<div
  className="navbar"
>
  <ul
    className="navbar-links"
  >
    <li
      className="q-mark-pic"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M2 0a2 2 0 0 0-2 2v12a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2H2zm3.496 6.033a.237.237 0 0 1-.24-.247C5.35 4.091 6.737 3.5 8.005 3.5c1.396 0 2.672.73 2.672 2.24 0 1.08-.635 1.594-1.244 2.057-.737.559-1.01.768-1.01 1.486v.105a.25.25 0 0 1-.25.25h-.81a.25.25 0 0 1-.25-.246l-.004-.217c-.038-.927.495-1.498 1.168-1.987.59-.444.965-.736.965-1.371 0-.825-.628-1.168-1.314-1.168-.803 0-1.253.478-1.342 1.134-.018.137-.128.25-.266.25h-.825zm2.325 6.443c-.584 0-1.009-.394-1.009-.927 0-.552.425-.94 1.01-.94.609 0 1.028.388 1.028.94 0 .533-.42.927-1.029.927z"
        />
      </svg>
    </li>
    <li
      className="navbar-links-item"
    >
      <a
        className="navbar-links-link"
        href="/"
        onClick={[Function]}
      >
        Home
         
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 576 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z"
          />
        </svg>
      </a>
    </li>
    <li
      className="navbar-links-item"
    >
      <a
        className="navbar-links-link"
        href="/take-a-quiz"
        onClick={[Function]}
      >
        Take Quiz 
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M505.12019,19.09375c-1.18945-5.53125-6.65819-11-12.207-12.1875C460.716,0,435.507,0,410.40747,0,307.17523,0,245.26909,55.20312,199.05238,128H94.83772c-16.34763.01562-35.55658,11.875-42.88664,26.48438L2.51562,253.29688A28.4,28.4,0,0,0,0,264a24.00867,24.00867,0,0,0,24.00582,24H127.81618l-22.47457,22.46875c-11.36521,11.36133-12.99607,32.25781,0,45.25L156.24582,406.625c11.15623,11.1875,32.15619,13.15625,45.27726,0l22.47457-22.46875V488a24.00867,24.00867,0,0,0,24.00581,24,28.55934,28.55934,0,0,0,10.707-2.51562l98.72834-49.39063c14.62888-7.29687,26.50776-26.5,26.50776-42.85937V312.79688c72.59753-46.3125,128.03493-108.40626,128.03493-211.09376C512.07526,76.5,512.07526,51.29688,505.12019,19.09375ZM384.04033,168A40,40,0,1,1,424.05,128,40.02322,40.02322,0,0,1,384.04033,168Z"
          />
        </svg>
      </a>
    </li>
    <li
      className="navbar-links-item"
    >
      <a
        className="navbar-links-link"
        href="/saved-quizzes"
        onClick={[Function]}
      >
        Saved Quizzes 
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M433.941 129.941l-83.882-83.882A48 48 0 0 0 316.118 32H48C21.49 32 0 53.49 0 80v352c0 26.51 21.49 48 48 48h352c26.51 0 48-21.49 48-48V163.882a48 48 0 0 0-14.059-33.941zM224 416c-35.346 0-64-28.654-64-64 0-35.346 28.654-64 64-64s64 28.654 64 64c0 35.346-28.654 64-64 64zm96-304.52V212c0 6.627-5.373 12-12 12H76c-6.627 0-12-5.373-12-12V108c0-6.627 5.373-12 12-12h228.52c3.183 0 6.235 1.264 8.485 3.515l3.48 3.48A11.996 11.996 0 0 1 320 111.48z"
          />
        </svg>
      </a>
    </li>
    <li
      className="navbar-links-item"
    >
      <a
        className="navbar-links-link"
        href="/random-quiz"
        onClick={[Function]}
      >
        Random Quiz 
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m13.293 2.707.818.818L3.318 14.318C2.468 15.168 2 16.298 2 17.5s.468 2.332 1.318 3.183C4.169 21.532 5.299 22 6.5 22s2.331-.468 3.182-1.318L20.475 9.889l.818.818 1.414-1.414-8-8-1.414 1.414zm3.182 8.354-2.403-2.404-1.414 1.414 2.403 2.404-1.414 1.415-.99-.99-1.414 1.414.99.99-1.415 1.415-2.403-2.404L7 15.728l2.403 2.404-1.136 1.136c-.945.944-2.59.944-3.535 0C4.26 18.795 4 18.168 4 17.5s.26-1.295.732-1.768L15.525 4.939l3.535 3.535-2.585 2.587z"
          />
        </svg>
      </a>
    </li>
    <li
      className="navbar-links-item"
    >
      <a
        className="navbar-links-link"
        href="/leaderboard"
        onClick={[Function]}
      >
        Leaderboard 
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M7.5 21H2V9h5.5v12zm7.25-18h-5.5v18h5.5V3zM22 11h-5.5v10H22V11z"
          />
        </svg>
      </a>
    </li>
  </ul>
</div>
`;
